test/integration/users_signup_test.rb

```diff
@@ -22,5 +22,8 @@ class UsersSignupTest < ActionDispatch::IntegrationTest
                                             password_confirmation: "password" }
     end
     assert_template 'users/show'
+    assert_select 'div#<CSS id for error explanation>'
+    assert_select 'div.<CSS class for field with error>'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.